src/structures/team_scoreboard.ts

```diff
@@ -0,0 +1,167 @@
+import Scoreboard from "./scoreboard";
+import Player from "./player";
+import Team from "./team";
+import _logger from "../logger";
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const logger = _logger("team_scoreboard");
+
+interface TeamMap {
+    [teamName: string]: Team;
+}
+
+export default class TeamScoreboard extends Scoreboard {
+    /**
+    * Mapping of team ID to Team
+    * Note: Each "player" in this.players represents a team
+    */
+    protected players: TeamMap;
```

Comment:
Since the base class has this type as protected, this one has to be protected too. This is the error:

 Class 'TeamScoreboard' incorrectly extends base class 'Scoreboard'.
  Property 'players' is private in type 'TeamScoreboard' but not in type 'Scoreboard'.